fix: Item configure

marination opened the pull request:
Port of https://github.com/frappe/erpnext/pull/22934 and https://github.com/frappe/erpnext/pull/22935

Another Fix: The product configurator would not show "Add to Cart" if product info is null. This PR overrides the effects of new changes in the configurator and shows the link anyway

scmmishra committed:
fix: unbound error in product configurator

scmmishra committed:
fix: handle product_info null

scmmishra commented:
@Mergifyio backport version-12-hotfix version-13-beta version-13-beta-pre-release v12-pre-release

marination committed:
fix: product action instead of info

marination committed:
fix: Show Add to Cart Button anyway

- add to cart wasnt visible if show price is disabled in shopping cart settings (product_info = null), creating confusion
